pkg/controller/tls/certs.go

```diff
@@ -225,55 +225,64 @@ func findExistingCertSecret(ctx context.Context, client kclient.Client, target s
+		logrus.Errorf("Error getting certificate secret %s/%s: %v", namespace, secretName, findSecretErr)
 		return findSecretErr
 	}
 
-	// Let's see if we have some existing certificate that matches the domain
-	existingSecret, err := findExistingCertSecret(ctx, client, domain)
-	if err != nil {
-		return err
-	}
-	if existingSecret != nil {
-		// We found an existing certificate
-		// 1. It's in the same namespace, so it will be picked up automatically
-		if existingSecret.Namespace == namespace {
-			logrus.Debugf("Found existing TLS secret %s/%s for domain %s", existingSecret.Namespace, existingSecret.Name, domain)
-			// Skip: TLS secret already exists, nothing to do here, renewal will be handled elsewhere
-			return nil
+	if !mustUpdate {
+		// Let's see if we have some existing certificate that matches the domain
+		existingSecret, err := findExistingCertSecret(ctx, client, domain)
+		if err != nil {
+			return err
 		}
+		if existingSecret != nil {
+			// We found an existing certificate
+			// 1. It's in the same namespace, so it will be picked up automatically
+			if existingSecret.Namespace == namespace {
+				logrus.Debugf("Found existing TLS secret %s/%s for domain %s", existingSecret.Namespace, existingSecret.Name, domain)
+				// Skip: TLS secret already exists, nothing to do here, renewal will be handled elsewhere
+				return nil
+			}
 
-		logrus.Debugf("Found existing TLS secret %s/%s for domain %s, copying to %s/%s", existingSecret.Namespace, existingSecret.Name, domain, namespace, secretName)
-
-		// 2. Copy secret to new namespace
-		copiedSecret := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretName,
-				Namespace: namespace,
-				Labels: map[string]string{
-					labels.AcornManaged: "true",
-				},
-				Annotations: map[string]string{
-					labels.AcornDomain: domain,
+			logrus.Debugf("Found existing TLS secret %s/%s for domain %s, copying to %s/%s", existingSecret.Namespace, existingSecret.Name, domain, namespace, secretName)
+
+			// 2. Copy secret to new namespace
+			copiedSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: namespace,
+					Labels: map[string]string{
+						labels.AcornManaged: "true",
+					},
+					Annotations: map[string]string{
+						labels.AcornDomain: domain,
+					},
 				},
-			},
-			Data: existingSecret.Data,
-			Type: existingSecret.Type,
-		}
+				Data: existingSecret.Data,
+				Type: existingSecret.Type,
+			}
 
-		if err := client.Create(ctx, copiedSecret); err != nil {
-			logrus.Errorf("Error creating TLS secret %s/%s: %v", copiedSecret.Namespace, copiedSecret.Name, err)
-			return err
+			if err := client.Create(ctx, copiedSecret); err != nil {
+				logrus.Errorf("Error creating TLS secret %s/%s: %v", copiedSecret.Namespace, copiedSecret.Name, err)
+				return err
+			}
+			return nil
 		}
-		return nil
 	}
 
 	go func() {
```

Comment:
Unrelated to the changes here, but can you help me understand why we do the update asynchronously here? Is it just for performance or is there something else?